backend/src/matches/matches.controller.spec.ts

```diff
@@ -24,12 +25,16 @@ describe('MatchesController', () => {
       controllers: [MatchesController],
       providers: [
         MatchesService,
+        {
+          provide: getRepositoryToken(Match),
+          useValue: jest.fn(),
+        },
         {
           provide: getRepositoryToken(User),
           useValue: jest.fn(),
         },
         {
-          provide: getRepositoryToken(Match),
+          provide: getRepositoryToken(MatchDto),
```

Comment:
This provide might not be necessary because of the additions at lines 28-31. I see that before there was the `getRepositoryToken` for the Dto, but I believe it should usually be the Entity itself (as you've done above).

@fyusuf-a knows what's going on with Match better than I do, so maybe he can correct me. But it seems strange. I'm wondering if the references to `Match` below should stay `Match` (meaning before they referenced the Dto but now they would reference the Entity)

Reply:
I think lines 31-39 do not make sense, there is already a `Match` entity repository. I think they should be removed.